QRCoder/PayloadGenerator.cs

```diff
@@ -65,6 +65,18 @@ private static bool IsValidQRIban(string iban)
     private static bool IsValidBic(string bic)
         => Regex.IsMatch(bic.Replace(" ", ""), @"^([a-zA-Z]{4}[a-zA-Z]{2}[a-zA-Z0-9]{2}([a-zA-Z0-9]{3})?)$");
 
+    /// <summary>
+    /// Validates the structure of a BIC with optional requirement check.
+    /// </summary>
+    /// <param name="bic">The BIC to validate.</param>
+    /// <param name="required">Whether the BIC is required. If false, null/empty values are considered valid.</param>
+    /// <returns>True if the BIC is valid; otherwise, false.</returns>
+    private static bool IsValidBic(string? bic, bool required)
+    {
+        if (string.IsNullOrEmpty(bic))
+            return !required;
+        return IsValidBic(bic!);
```

Comment:
```suggestion
        return IsValidBic(bic);
```
Null analysis in Roslyn should know that at this point `bic` isn't null.

Reply:
Unfortunately when targeting .net 3.5 and 4.0 the attributes are missing from IsNullOrEmpty so the compiler doesn't have that guarantee and the ! is necessary. (We have our own IsNullOrWhiteSpace function defined as a string extension to overcome this issue, but not IsNullOrEmpty.)

Reply:
Ah, I forgot that the `[NotNullWhen(false)]` attribute isn't there on older targets and that the compiler uses -- of course -- that ref-assemblies. Thanks for the reminder!